tests/org/mozilla/gc/TestRuntime.java

```diff
@@ -0,0 +1,37 @@
+package org.mozilla.gc;
+
+import gnu.testlet.TestHarness;
+import gnu.testlet.Testlet;
+
+public class TestRuntime implements Testlet {
+    public int getExpectedPass() { return 5; }
+    public int getExpectedFail() { return 0; }
+    public int getExpectedKnownFail() { return 0; }
+
+    public void test(TestHarness th) {
+        long totalMemory = Runtime.getRuntime().totalMemory();
+        long freeMemory = Runtime.getRuntime().freeMemory();
+
+        System.out.println("freeMemory1: " + Runtime.getRuntime().freeMemory());
+
+        th.check(totalMemory > freeMemory, "Total memory is strictly greater than free memory");
+        th.check(freeMemory > 0, "Free memory is strictly greater than 0");
+
+        for (int i = 0; i < 1000; i++) {
+            long[] array = new long[100000];
+        }
+
+        System.out.println("freeMemory2: " + Runtime.getRuntime().freeMemory());
+
+        th.check(totalMemory, Runtime.getRuntime().totalMemory(), "Total memory doesn't change");
+        th.check(freeMemory > Runtime.getRuntime().freeMemory(), "Free memory decreases after allocating some objects");
```

Comment:
Since *array* is out of scope by this assertion, couldn't its memory have been collected already? If so, then this doesn't necessarily check that *array* allocation decreases free memory. So I would put the assertion in a block with the allocation:

```Java
{
  long[] array = new long[100000];
  th.check(freeMemory > Runtime.getRuntime().freeMemory(), "Free memory decreases after allocating some objects"); 
}
```

(Unsure if you need to allocate repeatedly, but if so, then that block could be the existing *for* loop, in which case the assertion would run for every iteration of the loop.)